### PR TITLE
V2 client interaction

### PR DESCRIPTION
**Pull request by HelloGrayson**
Rework based on #29 
- [x] switch health ping to `HEAD` request
- [x] stream results to console
- [x] parse json from clients
- [x] clients return sub results
- [x] dont stop on first failure
- [x] exit with non-0 if any tests failed
- [x] timeout calls to client (created #26)
- [x] get basic output story in place
- [x] update docs (created #34)
- [x] yarpc-go `crossdock-v2` pr based on `crossdock:v2` - https://github.com/yarpc/yarpc-go/pull/54
- [x] yarpc-python `crossdock-v2` PR based on `crossdock:v2` - https://github.com/yarpc/yarpc-python/pull/28
- [x] yarpc-java `crossdock-v2` PR based on `crossdock:v2`
- [x] yarpc-node `crossdock-v2` PR based on `crossdock:v2`
- [x] rebase

Current output:

```
Crossdock starting...

Waiting on CROSSDOCK_CLIENTS=[alpha omega]

HTTP: Service omega:8080 is up
HTTP: Service alpha:8080 is up

All services are up after 202.128863ms!

Executing Matrix...

PASSED - {alpha map[behavior:dance]} - sup
PASSED - {alpha map[behavior:sing]} - sup
SKIPPED - {omega map[behavior:dance]} - sup
SKIPPED - {omega map[behavior:sing]} - sup

Tests passed!
```

cc @yarpc/yarpc

[Review on Reviewable](https://reviewable.io/reviews/yarpc/crossdock/30)

### CODE REVIEW

**Comment by HelloGrayson** [comment]
@kriskowal @junchaowu I've created a `crossdock-v2` branch & PR in Python and Go that uses this branch - can you guys do the same for Node and Java? I'd like to get on to v2 before too long so we know when tests fail or not.

**Comment by HelloGrayson** [comment]
@abhinav @kriskowal @junchaowu this is ready for a review.

**Comment by abhinav** [comment]
Reviewed 11 of 24 files at r1, 14 of 14 files at r2.
Review status: all files reviewed at latest revision, 8 unresolved discussions.

---

_[execute/entities.go, line 12 [r2]](https://reviewable.io:443/reviews/yarpc/crossdock/30#-KBtLsOo18ysdluYsq2N:-KBtLsOpcJwYJ_fxFhAz:-543564406) ([raw file](https://github.com/yarpc/crossdock/blob/8fcc8bb88b9f581094ba6224cb020a4831b226bc/execute/entities.go#L12)):_
Can you rename this pair to `TestResult` or `TestResponse` and `SubResult`

---

_[execute/entities.go, line 12 [r2]](https://reviewable.io:443/reviews/yarpc/crossdock/30#-KBtM87dnxVtjPdnQLiZ:-KBtM87dnxVtjPdnQLi_:441821216) ([raw file](https://github.com/yarpc/crossdock/blob/8fcc8bb88b9f581094ba6224cb020a4831b226bc/execute/entities.go#L12)):_
Also the comments could explain the relationship between these two types and TestCases better

---

_[execute/run.go, line 15 [r2]](https://reviewable.io:443/reviews/yarpc/crossdock/30#-KBtNQ4v3sErvMv-ust2:-KBtNQ4v3sErvMv-ust3:1966097114) ([raw file](https://github.com/yarpc/crossdock/blob/8fcc8bb88b9f581094ba6224cb020a4831b226bc/execute/run.go#L15)):_
Move this to a constant please

---

_[execute/run.go, line 32 [r2]](https://reviewable.io:443/reviews/yarpc/crossdock/30#-KBtPO_LyXPnBDcIWS2q:-KBtPO_LyXPnBDcIWS2r:694017295) ([raw file](https://github.com/yarpc/crossdock/blob/8fcc8bb88b9f581094ba6224cb020a4831b226bc/execute/run.go#L32)):_
Perhaps change to: "err: request failed: %v"

---

_[execute/run.go, line 43 [r2]](https://reviewable.io:443/reviews/yarpc/crossdock/30#-KBtPS5U5qd9stjDouPb:-KBtPS5U5qd9stjDouPc:1326061334) ([raw file](https://github.com/yarpc/crossdock/blob/8fcc8bb88b9f581094ba6224cb020a4831b226bc/execute/run.go#L43)):_
Same. Consider changing to "err: failed to parse response: %v"

---

_[execute/run.go, line 55 [r2]](https://reviewable.io:443/reviews/yarpc/crossdock/30#-KBtQFPsGBchUSsmlK89:-KBtQFPsGBchUSsmlK8A:2068611164) ([raw file](https://github.com/yarpc/crossdock/blob/8fcc8bb88b9f581094ba6224cb020a4831b226bc/execute/run.go#L55)):_
Consider moving the `8080` into a constant.

---

_[execute/run.go, line 80 [r2]](https://reviewable.io:443/reviews/yarpc/crossdock/30#-KBtQQ6ryu3feeTPhTdy:-KBtQQ6ryu3feeTPhTdz:65843982) ([raw file](https://github.com/yarpc/crossdock/blob/8fcc8bb88b9f581094ba6224cb020a4831b226bc/execute/run.go#L80)):_
Just return the `[]byte` here. You're converting the response back into `[]byte` in the caller to parse the JSON anyway.

---

_[execute/run_test.go, line 1 [r2]](https://reviewable.io:443/reviews/yarpc/crossdock/30#-KBtQeRrkInY1e_uHcwC:-KBtQeRrkInY1e_uHcwD:166874) ([raw file](https://github.com/yarpc/crossdock/blob/8fcc8bb88b9f581094ba6224cb020a4831b226bc/execute/run_test.go#L1)):_
?

---

_Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/yarpc/crossdock/30)_

**Comment by abhinav** [comment]
Reviewed 1 of 24 files at r1.
Review status: all files reviewed at latest revision, 8 unresolved discussions.

---

_Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/yarpc/crossdock/30)_

**Comment by kriskowal** [comment]
:lgtm:

---

Review status: 16 of 21 files reviewed at latest revision, 8 unresolved discussions.

---

_Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/yarpc/crossdock/30#-:-KCHyD3FEK9HumI5xcd1:1417088721)_

File: execute/run_test.go [hunk 1]
```diff
@@ -0,0 +1 @@
+package execute
```
Comment (abhinav) [inline execute/run_test.go:1]: ?

File: execute/wait.go [hunk 2]
```diff
@@ -37,7 +37,7 @@ func Wait(hosts []string, timeout time.Duration) {
 }
 
 // WaitForHTTPRequest polls host until it can make a request
-func WaitForHTTPRequest(host string, cancel <-chan struct{}) {
+func waitForHTTPRequest(host string, cancel <-chan struct{}) {
```
Comment (abhinav) [inline execute/wait.go:40]: update casing in comment

**Pull request merged**